<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>852</width>
    <height>637</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Algorithmic Trading</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_13">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="tabPosition">
       <enum>QTabWidget::North</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="interpolation_tab">
       <attribute name="title">
        <string>Interpolation</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Points</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <widget class="QSpinBox" name="interpolation_points_spin_box">
               <property name="maximum">
                <number>16777215</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_5">
            <property name="title">
             <string>Newton</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_6">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>Degree</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="interpolation_degree_spin_box">
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>50</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="interpolation_newton_plot_button">
               <property name="text">
                <string>Proceed</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_6">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_7">
            <property name="title">
             <string>Cube Spline</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <item>
              <widget class="QPushButton" name="interpolation_spline_plot_button">
               <property name="text">
                <string>Proceed</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_6">
            <property name="title">
             <string>Search</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <item>
              <widget class="QDateEdit" name="interpolation_date_edit"/>
             </item>
             <item>
              <widget class="QPushButton" name="interpolation_search_button">
               <property name="text">
                <string>Get values</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>17</width>
              <height>37</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Plot" name="interpolation_plot" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="approximation_tab">
       <attribute name="title">
        <string>Approximation</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_12">
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string>Points</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QSpinBox" name="approximation_points_spin_box">
               <property name="maximum">
                <number>16777215</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_9">
            <property name="title">
             <string>Degree</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_10">
             <item>
              <widget class="QSpinBox" name="approximation_degree_spin_box">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>50</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_10">
            <property name="title">
             <string>Period</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_11">
             <item>
              <widget class="QSpinBox" name="period_spin_box">
               <property name="suffix">
                <string> d</string>
               </property>
               <property name="value">
                <number>14</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="approximation_plot_button">
            <property name="text">
             <string>Proceed</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_8">
            <property name="title">
             <string>Search</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <item>
              <widget class="QDateEdit" name="approximation_date_edit"/>
             </item>
             <item>
              <widget class="QPushButton" name="approximation_search_button">
               <property name="text">
                <string>Get value</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_11">
            <property name="title">
             <string>Research</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_15">
             <item>
              <widget class="QPushButton" name="approximation_research_button">
               <property name="text">
                <string>Proceed</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>214</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Plot" name="approximation_plot" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="research_tab">
       <attribute name="title">
        <string>Research</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_14">
          <item>
           <widget class="QGroupBox" name="groupBox_4">
            <property name="title">
             <string>Points</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QSpinBox" name="research_points_spin_box">
               <property name="maximum">
                <number>16777215</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="title">
             <string>Partitions</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QSpinBox" name="research_partitions_spin_box">
               <property name="minimum">
                <number>1</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="interpolation_research_button">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Proceed</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>179</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Plot" name="researchPlot" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>852</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuPlot">
    <property name="title">
     <string>Plot</string>
    </property>
    <addaction name="actionClear"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuPlot"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clear</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Plot</class>
   <extends>QWidget</extends>
   <header>plot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnActionOpenTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClear</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnActionClearTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnActionQuitTriggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>interpolation_newton_plot_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnInterpolationNewtonPlotButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>206</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>interpolation_spline_plot_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnInterpolationSplinePlotButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>281</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>approximation_plot_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnApproximationPlotButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>278</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>interpolation_search_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnInterpolationSearchButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>384</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>approximation_search_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnApproximationSearchButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>371</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>interpolation_research_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnInterpolationResearchButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>65</x>
     <y>205</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>approximation_research_button</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OnApproximationResearchButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>437</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OnActionOpenTriggered()</slot>
  <slot>OnActionClearTriggered()</slot>
  <slot>OnActionQuitTriggered()</slot>
  <slot>OnInterpolationNewtonPlotButtonClicked()</slot>
  <slot>OnInterpolationSplinePlotButtonClicked()</slot>
  <slot>OnApproximationPlotButtonClicked()</slot>
  <slot>OnInterpolationSearchButtonClicked()</slot>
  <slot>OnApproximationSearchButtonClicked()</slot>
  <slot>OnInterpolationResearchButtonClicked()</slot>
  <slot>OnApproximationResearchButtonClicked()</slot>
 </slots>
</ui>
